### ISSUE TITLE
Feature request: Changelog

### ISSUE THREAD

**Issue by raRaRa** [blueimp/JavaScript-Load-Image#91]
The title says it all! It would be really great to have a changelog to know what has changed before upgrading.

Thanks!

**Comment by blueimp** [comment]
Hi @raRaRa,

while I don't keep a manual changelog, I generally try to write useful commit messages, so the [commit history](https://github.com/blueimp/JavaScript-Load-Image/commits/master) is the closest thing to a formal changelog.

The following is the output of `git log --pretty=%B master | cat`:

```
2.26.0

Run mocha tests via Docker.

2.25.0

Add loading images via URL support to the demo.

Add XHR2 fallback if fetch API is not available.

This adds support to parse meta data for images loaded via URL for Internet Explorer 10+.

Check if blob argument is a Blob instance.

Update dev dependencies.

Add GitHub Sponsors config.

Run workflow on push and pull_request.

Format YAML with prettier.

Remove obsolete travis config.

Add GitHub NodeJS workflow.
2.24.0

Update dev dependencies.

2.23.0

Update test vendor dependencies.

Rebuild minified bundle.

Use blueimp+jsdoc+prettier config for eslint.

Move eslint configs to package.json.
Use package.json files property instead of .npmignore.

Format markdown using prettier.

Reformat license text.

2.22.0

Update dev dependencies.

2.21.0

Provide original image width+height in callback data.

Closes #90

Remove unnecessary format script.

Update devDependencies.

Simplify eslintrc.js.

Use correct corresponding closing tags.

2.20.1

Validate IPTC segment offsets.

2.20.0

Cleanup.
- Add APP13 marker list to meta plugin.
- Add IPTC source files to index.js.
- Format source code files (HTML+CSS+JS) files with Prettier.
- Keep JS source code inside of 80 column limit.
- Merge the separate IPTC demo into the main demo.
- Add a button to cancel image cropping.
- Include orientation plugin before Exif/IPTC plugins.
- Write IPTC capitalized except for variable and function names.
- Update project description.
- Mention Dave Bevan's IPTC contribution in credits section.

Merge pull request #89 from bevand10/master

Add an IPTC parser
Expand auto-header parse to include Iptc, expand test description string to suit.

Add a basic Iptc test.

Include the new iptc parser files.

Add iptc files to min package, add Iptc to description and tags.

Initial test update - add Iptc data (ObjectName 2:5) to the test image.

Hopefully the final Travis fix!

More Travis fixes

Remove errant , and clean up record of IPTC fields we do not read

Remove errant ,

Switch quotes

Fix JS formatting issues thrown by Travis CI

Revise the 8BIM detection method to correctly parse the three std IPTC test images at https://iptc.org/standards/photo-metadata/iptc-standard/

Add copyright and IPTC source references.

Add an IPTC parser, plus documentation, demo page etc.

2.19.0

Use loadImage.createObjectURL.
This ensures compatibility with older browsers, see:
https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL#Browser_compatibility

Merge pull request #88 from bubbatls/patch-1

Optimisation for thumbnail
use better javascript
Optimisation for thumbnail

There is no need to rebuild a data uri.
We can use URL.createObjectURL and a Blob.
It is faster, and further use of the URL will be too.
Using sudo as workaround for https://github.com/shellscape/mocha-chrome/pull/21

2.18.0

Update jQuery version.

2.17.1

Minor demo fixes.
- Export to original type if possible.
- Update download file extension if export type differs.

2.17.0

Format code with prettier-eslint.

2.16.0

Ensure chrome headless is available in travis.

2.15.0

Use standard via eslint config.
Use chrome and a local webserver for the unit tests.
Update uglifyjs settings.

Update canvas.toBlob() polyfill.

2.14.0

Update canvas-to-blob test library.

Add license file.

2.13.1

Fix issue with bundlers that wrap code (e.g. browserify).

2.13.0

Use $ instead of window for consistency.

Merge pull request #86 from gre/master

Fix import from Node environment (to not break in server side rendering)
use $

Fix import from Node environment (to not break in server side rendering)

2.12.2

Include the scale and fetch plugins in the module.

2.12.1

Add the fetch plugin to the minified output.

2.12.0

Fetch blob from URL if meta data should be retrieved.
This feature requires browser support for the fetch API.

Make argument to hasMetaOption optional.

Load meta library before orientation.

Move image scaling functionality into a separate module.

2.11.0

Update license URL.

2.10.1

Merge pull request #85 from fleg/master

fix crop with downsamplingRatio
Fix crop with downsamplingRatio

2.10.0

Also include the minified version in the npm package.

Move demo script into separate folder.

2.9.0

Add .npmignore.

Move index.js into js directory.

Merge pull request #84 from redgeoff/patch-1

fix(package): main should reference index.js
fix(package): main should reference index.js
2.8.1

Add test case for pull request #83.

Merge pull request #83 from poshaughnessy/master

Fix incorrect maxWidth/maxHeight when rotated
Fix incorrect maxWidth/maxHeight when rotated

2.8.0

Add main file to package.json.

2.7.0

Add options meta:true and orientation:true to the documentation.

Update mocha to version 3.1.0.

Replace expect.js with chai.js.

Add auto-orient option via options.orientation === true.

Add option to load image meta data automatically on load.

Use revoke helper function.

Remove obsolete _type property.

Delete _objectURL property after revoke.

Also revoke objectURLs on error.

Make img event handlers extensible.

Fix linting issues.

Update dev dependencies.

Merge pull request #82 from Nateowami/master

Update README.md to clarify specifying orientation from Exif data
Update README.md to clarify specifying orientation from Exif data

2.6.2

Don't throw for invalid EXIF values.
Fixes #81

2.6.1

StandardJS compatibility.

Revert premature version change.

Merge pull request #75 from micky2be/master

Webpack support
Merge branch 'Brogency-master'

Merge branch 'master' of git://github.com/Brogency/JavaScript-Load-Image into Brogency-master

2.6.0

Add Travis CI config file.

2.5.0

Remove the obsolete iOS scaling fixes.
The related issues have been resolved in WebKit for iOS 8:
http://trac.webkit.org/changeset/170675

2.4.0

Add downsamplingRatio option.

2.3.1

Add pixelRatio option.
Adjust orientation with enabled pixelRatio.
Update tests.

2.3.0

Merge upscale canvas drawing on retina and update tests.

2.2.0

Use JavaScript Standard Style.

Downsample in 2x steps for better image quality

See http://stackoverflow.com/questions/17861447/html5-canvas-drawimage-how-to-apply-antialiasing
Changed paths for define

Added require support

upscale canvas drawing on retina

2.1.0

Add source map for the minified JS files.

2.0.1

Keep the gh-pages branch in sync with master.

2.0.0

Better support Node.js style module loaders.

Copy vendor dependencies into project source.

Remove obsolete bower config file.

Remove obsolete Grunt dependencies.
Use mocha-phantomjs to run the unit tests in the console.
Update dev dependencies.

Remove obsolete file version numbers.

Use window object directly to support browserify.

Add license field to package definition(s).

Renamed combined and minified file to load-image.all.min.js. Closes #63

Mocha tests require the url protocol.

Use https for the demo url.

Added aspectRatio option. Closes #50

Ignore too large orientation values. Fixes #56

Rounding error patch release.

Updated devDependencies.

Removed Math.ceil calls to avoid rounding errors.

Let the browser implicitly cast calculated image dimensions from
floating point values to integer values.

Handle FileReader errors.

Added grunt-mocha task.

Replaced Makefile with Grunt; only distribute relevant files via bower.

JSHint compatibility update.

Hide URL/FileReader API requirement message in capable browsers. Closes #35

Fix scaling images on iOS which are not subsampled, but do suffer a vertical squash.

Added sourceWidth, sourceHeight, top, right, bottom and left options to allow arbitrary cropping. Closes #31

Moved orientation functionality into a separate file.
Added the Jcrop plugin to the demo to visualize the new cropping
features.

Fixed cropping with top margin and vertical squash.

Fixed cropping for subsampled images.

Added contain and cover options. Closes #30.

Updated path to canvas.toBlob polyfill library.

Refactored demo implementation.

Added bower.json.

Added missing tags for Exif version 2.3. Fixes #25.

Added comment for the maxMetaDataSize option.

Break out of the loop if the marker length exceeds the data view length.

Use 256 KiB by default to parse the image meta data, which can be
overriden by the new configuration option maxMetaDataSize.

Provide the loadImage.blobSlice method only if the browser supports slicing Blobs.

Added meta data tests.

Fixed demo error.

Added an extension to parse image meta data, including Exif tags and thumbnails.

Split the Load Image library into one main module and several
extensions.

Catch "No Exif data" errors.

Log parsed EXIF data.

Updated the minified version.

Added orientation tests.

Assume canvas:true if an orientation is given.

Updated package version.

Added orientation information to the options listing.

Updated the minified version.

Added an EXIF Reader library to demonstrate the orientation option.

Accept the reading method as third argument to loadImage.readFile.

Added orientation support.

Added crop option to enfore maxWidth & maxHeight constraints.

Return false if file parameter is not a File, Blob or String.

Fixed last merge.

Merge pull request #21 from phoenix303/master

Multiple errors for non blob files - fixed 
Multiple errors for non blob files - fixed

fixed multiple error prompts when uploading non blob files

Added crossOrigin to the options list.

Added crossOrigin option for CORS enabled images. Fixes #19.

Minor JSLint compatibility update for latest merge.

Merge pull request #17 from igorbernstein/extra_ie8_request

Be a little more defensive on types of inputs
if the file api is not supported & the url is not a string avoid a useless request for [object]

Use Math.ceil instead of bitwise left shift.

Width or height might exceed 2147483648 pixels, which results in
negative numbers when using bitwise shift.
Also, 1px scaled with a factor of less than 1 should still be 1px.

Update test page jQuery version.

Create minified script and update package version.

Fixed image resize algorithm for subsampled JPEGs.

Thanks to Claudio (bytepan) for his patch to the original
ios-imagefile-megapixel library:
https://github.com/bytepan/ios-imagefile-megapixel/commit/f58d6d65c828dd
5741edf30d561220b3de94c42e

Update demo jQuery version to 1.9.1.

Use the faster bitwise operator instead of Math.floor and parseInt.

See
https://github.com/stomita/ios-imagefile-megapixel/commit/a8a96e47bb25c6
ad59f84a9144a38198ce974cfc
and http://jsperf.com/math-floor-vs-math-round-vs-parseint/33

Prevent division by zero.

See
https://github.com/stomita/ios-imagefile-megapixel/commit/5b2252cc14d412
24682f06c098c1b44fef0258d5

Handle empty results and errors for images loaded via FileReader interface.

Fixed image scaling on iOS devices - Fixes #13.

iOS image scaling fixes based on
https://github.com/stomita/ios-imagefile-megapixel

Minor JSLint compatibility update.

Added X-UA-Compatible meta tag to force latest IE rendering engine or ChromeFrame if installed.

Refactor last merge and update version number.

Add tests for the noRevoke option.

Merge pull request #9 from joewood/master

Check for noRevoke breaks integration with jquery file upload.
Fix to check valid 'options' before noRevoke chk

Added a test to see if options is set before dereferencing the object to
check for noRevoke.

Fixed package version.

Updated README and package information.

Refactor last merge.

use "noRevoke" instead of "nounload".
Update Version number.
Update minified version.

Merge pull request #7 from royhaddad/patch-1

Added the option 'nounload'
Added the option 'nounload'

If the image data was loaded into an object URL, this option prevents it from
being unloaded once the image is ready.
Refactor last merge with a fix for Opera 12.

Update demo page to fix drag&drop for Opera 12.

Merge pull request #6 from naoina/master

Fixed a file loading on Opera browser
Fixed a file loading on Opera browser

Update tests for version 2.0 of the JavaScript Canvas to Blob function.

Accept a canvas element as parameter for loadImage.scale.

Only apply the new width/height after all calculations have been done.

Set viewport to device width.

Use a .gitignore file instead of the files field in package.json.

Remove the obsolete script file.

Use the Canvas to Blob API to create the Test Blob.

Use mocha+expect.js as testing framework.

Register as anonymous AMD module.

Added Makefile to create the minified version.
Updated Bootstrap version.

Fixed file loading on browsers supporting the File API but not Blobs (e.g. Firefox 3.6).

Moved Bootstrap IE6 fixes into a separate file.

Updated package information.

Updated license link and package information.

Add better capability checks for the Blob, FileReader and URL APIs.

Do not scale to max/min settings without min/max settings.

Fixed scaling algorithm for cases with both max and min settings.

Return the img element or FileReader object to allow aborting the image load.

Initial commit.
```

**Comment by blueimp** [comment]
Closing this as it's unlikely I'll adopt a manual change log.
Despite good reasons for having one it's simply a time commitment I'm not willing to invest for now.
